tdrs-backend/tdpservice/users/permissions.py

```diff
@@ -3,18 +3,19 @@
 from rest_framework import permissions
 
 
-class IsUserOrReadOnly(permissions.BasePermission):
+class IsUserOrAdmin(permissions.BasePermission):
     """Object-level permission to only allow owners of an object to edit it."""
 
-    def has_permission(self, request, view):
-        """Check if user has required permissions."""
-        if request.method in permissions.SAFE_METHODS:
-            return True
-        return request.user.is_authenticated
-
     def has_object_permission(self, request, view, obj):
         """Check if user has required permissions."""
-        if request.method in permissions.SAFE_METHODS:
-            return True
+        return obj == request.user or (
```

Comment:
@carltonsmith What is the left-hand `request.user` side of the or statement doing here? I read it as saying that any user will have a `IsUserOrAdmin` value of `true`, since `request.user` can be truthy even if the user is not authenticated or not an admin.

Reply:
@alexsoble I can see how it would read that way, but that's actually not what's happening here. This is a specific method of the Django Rest Framework to check on permissions for the object defined in the app. Since this `permissions.py` exist in the `users` app, the `obj` passed to this function is the user related to the object. So, say I am going to edit a user. When I do, this method will be passed an `obj` argument that represents the user I am trying to access. The check of `if obj == request.user` is asking if the object I am requesting access to and the user making the request is the same. If it is, that means that the user is attempting to edit their own profile.

If we were to include a `permissions.py` in another object and check for something similar, we could attach an `owner` to the `obj` and then check for `if obj.owner == request.user` the same way, but since the `obj` itself is the user here, this is the way it works.

Reply:
I see! Thanks for the explanation. So is this in reference to any ACs or use cases that we want to build out, or is more about preparing for future functionality in general?

Reply:
This is more about shoring up proper permissions for existing functionality. When we implemented creating users we hadn't implemented permissions yet. This makes sure what we already have is secured properly with the new functionality this PR adds.